src/mapgen_valleys.cpp

```diff
@@ -521,35 +471,18 @@ float MapgenValleys::terrainLevelFromNoise(TerrainNoise *tn)
 	*tn->rivers = base;
 	if (river < 0.f) {
 		// Use the the function −sqrt(1−x²) which models a circle.
-		float depth = (river_depth * sqrt(1.f - pow((river / river_size + 1), 2)));
+		float depth = (river_depth_bed
+				* sqrt(1.f - pow((river / river_size_factor + 1), 2)));
 
 		// base - depth : height of the bottom of the river
-		// water_level - 2 : don't make rivers below 2 nodes under the surface
-		u16 min_bottom = 2;
-		if (!fast_terrain)
-			min_bottom = 6;
-		mount = fmin(fmax(base - depth, (float) (water_level - min_bottom)), mount);
+		// water_level - 6 : don't make rivers below 6 nodes under the surface
+		u16 min_bottom = 6;
+		mount = MYMIN(MYMAX(base - depth, (float) (water_level - min_bottom)), mount);
```

Comment:
Expressions like MYMIN(MYMAX(x, y), z) could be simplified to rangelim(x, y, z).